CANTroller2/src/qpid.h

```diff
@@ -447,12 +436,153 @@ float QPID::GetOutputSum() { return outputSum; }
 float QPID::GetOutputMin() { return outMin; }  // Soren
 float QPID::GetOutputMax() { return outMax; }  // Soren
 float QPID::GetOutputRange() { return outMax - outMin; }  // Soren
+float QPID::GetCenter() { return center; }  // Soren
 uint8_t QPID::GetMode() { return static_cast<uint8_t>(mode); }
 uint8_t QPID::GetDirection() { return static_cast<uint8_t>(action); }
 uint8_t QPID::GetPmode() { return static_cast<uint8_t>(pmode); }
 uint8_t QPID::GetDmode() { return static_cast<uint8_t>(dmode); }
 uint8_t QPID::GetAwMode() { return static_cast<uint8_t>(iawmode); }
 uint8_t QPID::GetCentMode() { return static_cast<uint8_t>(centmode); }  // Soren
-float QPID::GetCenter() { return center; }  // Soren
 
+class IdleControl {  // Soren - To allow creative control of PID targets in case your engine idle problems need that.
+  public:
+    enum class idlemodes : uint32_t { direct, control, minimize, num_idlemodes };  // direct: disable idle management.  control: soft landing to idle rpm.  minimize: attempt to minimize idle to edge of instability

```

Comment:
Using num_idlemodes is a clever way to get the count of items in the enum, eg IdleMode::Count, but be careful not to try to use it as an IdleMode eg like don't pass it to a function that needs a mode.